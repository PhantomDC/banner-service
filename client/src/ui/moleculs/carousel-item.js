import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import {
  FlexWrap,
} from '../'


const Image = FlexWrap.extend`
  background: url(${props => props.url});
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
`

const carouselItem = ({
  className, img, activeImage, ids, width, ...rest
}) => (
  <FlexWrap
    w={width}
    h="146px"
    ai="center"
    jc="center"
    className={`${className} ${activeImage === ids && 'active'}`}
    {...rest}
  >
    <FlexWrap
      className="percent"
      w="100%"
      h="100%"
      ai="center"
      jc="center"
    >
      {img.info && `-${img.info.percentCompress}%`}
    </FlexWrap>
    <Image
      w="90%"
      h="90%"
      url={`http://localhost:8000/decompress/${img.url}`}
    />
  </FlexWrap>
)

carouselItem.propTypes = {
  className: PropTypes.string.isRequired,
  img: PropTypes.shape({
    url: PropTypes.string,
    info: PropTypes.object,
  }).isRequired,
  activeImage: PropTypes.number,
  ids: PropTypes.number.isRequired,
  width: PropTypes.string.isRequired,
}

carouselItem.defaultProps = {
  activeImage: null,
}

export const CarouselItem = styled(carouselItem)`
  position: relative;
  background-color: #e3e3e3;
  border-radius: 4px;
  margin-right: 15px;
  cursor: pointer;
  box-shadow: 0 14px 20px 0px rgba(0,0,0,0.04), 0 10px 10px rgba(0,0,0,0.05);
  & .percent {
    position: absolute;
    color: #000000;
    font-size: 30px;
    font-weight: bold;
    z-index: 50;
    opacity: 0.6;
    text-shadow: -2px -2px 0 #fff, 2px -2px 0 #fff, -2px 2px 0 #fff, 2px 2px 0 #fff;
  }
  &.active {
    background: #c8c8c8 !important
  }
`
