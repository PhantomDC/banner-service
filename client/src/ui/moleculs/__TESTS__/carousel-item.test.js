import React from 'react'
import { configure, render, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { create } from 'react-test-renderer'
import { CarouselItem } from '../carousel-item'

/* eslint-disable no-undef */
/* eslint-disable function-paren-newline */

configure({ adapter: new Adapter() })

let props
let withInfo
let withoutInfo

beforeEach(() => {
  withInfo = {
    url: 'path_to_image',
    info: {
      percentCompress: 0,
    },
  }
  withoutInfo = {
    url: 'path_to_image',
    info: null,
  }
  props = (info = true) => ({
    activeImage: 1,
    ids: 0,
    width: '240px',
    img: info ? withInfo : withoutInfo,
  })
})

describe('test component CarouselItem', () => {
  it('CarouselItem render without crash', () => {
    render(<CarouselItem {...props()} />)
  })

  it('CarouselItem render with info', () => {
    const wrapper = shallow(<CarouselItem {...props()} />)
    const { img } = wrapper.instance().props

    expect(img).toBe(withInfo)
  })

  it('CarouselItem render without info', () => {
    const wrapper = shallow(<CarouselItem {...props(false)} />)
    const { img } = wrapper.instance().props

    expect(img).toBe(withoutInfo)
  })

  it('CarouselItem props is equals', () => {
    const data = props(false)
    const wrapper = shallow(<CarouselItem {...data} />)
    const { img, activeImage, width, ids } = wrapper.instance().props

    expect(img).toBe(data.img)
    expect(activeImage).toBe(data.activeImage)
    expect(width).toBe(data.width)
    expect(ids).toBe(data.ids)
  })

  it('shapshot Button', () => {
    const tree = create(<CarouselItem {...props()} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
