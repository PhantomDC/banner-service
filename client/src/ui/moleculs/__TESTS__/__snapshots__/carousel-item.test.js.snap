// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component CarouselItem shapshot Button 1`] = `
<div
  className="sc-dxgOiQ fUeCKG false sc-bxivhb eusAaK"
  height="146px"
  width="240px"
>
  <div
    className="percent sc-bxivhb ieqoeX"
    height="100%"
    width="100%"
  >
    -0%
  </div>
  <div
    className="sc-kpOJdX gyPaLp"
    height="90%"
    width="90%"
  />
</div>
`;
